src/Forms/GridField/GridField.php

```diff
@@ -113,21 +135,7 @@ class GridField extends FormField
      *
      * @var array
      */
-    protected $readonlyComponents = [
-        GridField_ActionMenu::class,
-        GridFieldConfig_RecordViewer::class,
-        GridFieldButtonRow::class,
-        GridFieldDataColumns::class,
-        GridFieldDetailForm::class,
-        GridFieldLazyLoader::class,
-        GridFieldPageCount::class,
-        GridFieldPaginator::class,
-        GridFieldFilterHeader::class,
-        GridFieldSortableHeader::class,
-        GridFieldToolbarHeader::class,
-        GridFieldViewButton::class,
-        GridState_Component::class,
-    ];
+    protected $readonlyComponents = [];
```

Comment:
Can you help me understand where this is used and why it's ok to remove all the entries here, or where/when they would matter and how they can be provided in another way? Thanks!

Reply:
That variable is only ever used in `getReadonlyComponents` and `setReadonlyComponents`. Instead of having it declared with a value as part of the class definition, we now call `setReadonlyComponents` on instantiation and pass through the new `$default_readonly_components` configuration variable, which allows the components to be defined in yml config instead of being effectively immutable at a global level.

The same set that used to be declared in that variable is now the default for `$default_readonly_components`.

Reply:
Right, I see. So in your custom config, do you exclude some classes from the read-only transformation? Or add more to it?

It nearly feels like everything should always be transformed, shouldn't it?

Reply:
I have some projects where I want to globally add to, remove, or replace components from that readonly configuration - and it is a pain trying to do that at the moment as the only way to interact with that config is on a per-instance basis. I haven't found a nice way to do it.

My most common use case is adding a new third-party component which I want to be included in all readonly gridfields.

Reply:
@michalkleiner does the above make sense? It'd be nice to see this go in 4.11.0 alongside #10204 since they're thematically related.